src/DynamoCore/Models/CustomNodeWorkspaceModel.cs

```diff
@@ -191,7 +191,9 @@ public override bool SaveAs(string newPath, ProtoCore.Core core)
             // custom node with a new function id
             if (originalPath != newPath)
             {
-                CustomNodeId = Guid.NewGuid();
+                // If it is a newly created node, no need to generate a new guid
+                if (!string.IsNullOrEmpty(originalPath))
+                    CustomNodeId = Guid.NewGuid();
```

Comment:
What about this scenario:

1. Save the custom workspace, `Guid01` is now the new ID.
2. Save the custom workspace as another file.
3. Sounds like the above step will not generate a new ID for the new DYF file.

Point number 3 is what the test was originally testing `Assert.AreNotEqual(initialId, newDef);`, sounds like it is expected to have a different `Guid` for `SaveAs`.